### PR TITLE
Closing dangling file pointers

### PR DESCRIPTION
**Pull request by stefdoerr**
Hi, I was getting lots of warnings about open file pointers so I thought I'd make this PR to resolve the warnings. Also I assume eventually you might hit OS issues if you open too many files while using propka programmatically through python.

### CODE REVIEW

File: propka/input.py [hunk 1]
```diff
@@ -135,11 +135,13 @@ def read_parameter_file(input_file, parameters):
...
     try:
         ifile = resource_filename(__name__, input_file)
-        input_ = open_file_for_reading(ifile)
+        input_, stream = open_file_for_reading(ifile)
     except (IOError, FileNotFoundError, ValueError, KeyError):
-        input_ = open_file_for_reading(input_file)
+        input_, stream = open_file_for_reading(input_file)
     for line in input_:
         parameters.parse_line(line)
+    if not stream:
```
Comment (sobolevnrm) [inline propka/input.py:143]: Couldn't this be done more cleanly by either checking for duck-typed attributes or simply using `isinstance`?

**Review by sobolevnrm — CHANGES_REQUESTED** [review]

**Comment by speleo3** [comment]
How about solving this with a context manager (and removing the ["TODO - convert this to a context manager"](https://github.com/jensengroup/propka/blob/68007f065c5400566598509875344ed1c0f593c9/propka/input.py#L22) comment)? If you don't mind, I can open a pull request with the solution I have in mind, but I'd also be happy to review yours if you want to give it a try.

**Comment by intendo** [comment]
> How about solving this with a context manager (and removing the ["TODO - convert this to a context manager"](https://github.com/jensengroup/propka/blob/68007f065c5400566598509875344ed1c0f593c9/propka/input.py#L22) comment)? If you don't mind, I can open a pull request with the solution I have in mind, but I'd also be happy to review yours if you want to give it a try.

I would vote for the context manager approach.

**Comment by stefdoerr** [comment]
Sure feel free to open another PR!

**Pull request closed without merging**